services/publishing/Tweek.Publishing.Service/Helpers/TriggerHooksHelper.cs

```diff
@@ -19,38 +21,77 @@ public class TriggerHooksHelper {
     private readonly MetricTags _metricsFailure = new MetricTags("Status", "Failure");
 
     public TriggerHooksHelper(HttpClient client, IMetrics metrics, ILogger logger = null) {
-      this._logger = logger ?? NullLogger.Instance;
-      this._client = client;
-      this._metrics = metrics;
+      _logger = logger ?? NullLogger.Instance;
+      _client = client;
+      _metrics = metrics;
     }
 
-    public async Task TriggerHooks(Dictionary<( string type, string url ), string> hooksWithData, string commitId) {
-      var triggerTasks = hooksWithData.Select( kvp => ( task: TriggerHook(kvp.Key, kvp.Value), hookType: kvp.Key.type ) );
-
-      foreach (var triggerTask in triggerTasks) {
+    public async Task TriggerHooks(Dictionary<Hook, HookData> hooksWithData, string commitId) {
+      var triggerTasks = hooksWithData.Select( kvp => ( task: TriggerHook(kvp.Key, kvp.Value), hookType: kvp.Key.Type ) );
+       
+      foreach (var (task, hookType) in triggerTasks) {
         try {
-          await triggerTask.task;
+          await task;
 
           _metrics.Measure.Counter.Increment(_hooksMetric, _metricsSuccess);
         } catch (Exception ex) {
-          _logger.LogError(ex, $"Failed triggering a hook of type {triggerTask.hookType} for commit {commitId}");
+          _logger.LogError(ex, $"Failed triggering a hook of type {hookType} for commit {commitId}");
           _metrics.Measure.Counter.Increment(_hooksMetric, _metricsFailure);
         }
       }
     }
 
-    private async Task TriggerHook(( string type, string url ) hook, string payload) {
-      switch (hook.type) {
-        case "notification_webhook":
-          await TriggerNotificationWebhook(hook.url, payload);
+    private async Task TriggerHook(Hook hook, HookData hookData)
+    {
+      switch (hook.Type) {
```

Comment:
It might be clearer and less verbose with a local dictionary of format functions instead of a switch